### PR TITLE
Use error types in RMB direct client

### PR DESCRIPTION
**Pull request by 0oM4R**
### Description

use types/errors in rmb direct client

### Changes

- added rmb error and connection error to types/errors
- use types/errors in rmb direct client f

### Related Issues
- #1417 
- #995 
### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/rmb_direct_client/lib/client.ts [hunk 1]
```diff
@@ -103,16 +104,17 @@ class Client {
...
       this.twin = await this.tfclient.twins.get({ id: twinId });
       if (!twinId) {
-        throw new Error({ message: `Couldn't find a user for the provided mnemonic on this network.` });
+        throw new ValidationError(`Couldn't find a user for the provided mnemonic on this network.`);
       }
     } catch (e) {
       console.log(e);
-      throw new Error({ message: `Couldn't find a user for the provided mnemonic on this network.` });
+      //TODO is this a valid error message?
+      throw new ConnectionError(`Couldn't find a user for the provided mnemonic on this network.`);
```
Comment (AlaaElattar) [inline packages/rmb_direct_client/lib/client.ts:112]: I think this is a validation Error not connection. What do u think ?

**Review by AlaaElattar — CHANGES_REQUESTED** [review]

File: packages/rmb_direct_client/lib/client.ts (hunk 1 above)
Comment (0oM4R) [reply packages/rmb_direct_client/lib/client.ts:112]: i think this one is the validation related 
https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1433/files/d251636ea203fb9362bd10a0efc8b6deac7d9203#diff-3290bb9d40c106d5afd7fbca7c8c3d3bc60d1025e6c5b6634f0e244138fd215cR107
 
but in catch from my usage in playground i got this error many times while network is offline so i mentioned in line 111 to check this error message

Comment (AlaaElattar) [reply packages/rmb_direct_client/lib/client.ts:112]: maybe we can change it to `Couldn't find a user for the provided mnemonic, check your connection`. or any msg with indication that the problem in the connection

Comment (AhmedHanafy725) [reply packages/rmb_direct_client/lib/client.ts:112]: IMO, this try-and-catch block was added for having a more user-friendly error. But now tfchain should throw the correct error with its type, no?

Comment (0oM4R) [reply packages/rmb_direct_client/lib/client.ts:112]: > IMO, this try-and-catch block was added for having a more user-friendly error. But now tfchain should throw the correct error with its type, no?

exactly but the question is what it the correct error in this case ?

**Comment by 0oM4R** [comment]
will draft this pr to user types in reject ;

File: packages/rmb_direct_client/lib/client.ts [hunk 2]
```diff
@@ -324,7 +322,8 @@ class Client {
             }
           } else {
             this.responses.delete(requestID);
-            reject("invalid signature, discarding response");
+            //TODO what is the probate error type
```
Comment (0oM4R) [inline packages/rmb_direct_client/lib/client.ts:325]: @AhmedHanafy725 what do you think

**Review by AlaaElattar — APPROVED** [review]

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**